src/clj_puppetdb/http.clj

```diff
@@ -24,47 +24,86 @@
   `puppetlabs.ssl-utils.core/pems->ssl-context` function."
   [:ssl-cert :ssl-key :ssl-ca-cert])
 
+(defn- make-client-common
```

Comment:
I really like this approach to the client.

+1 to Karel's comment on vcr separation. Way better than before.